BridgeAppSDK/SBAProfileItem.swift

```diff
@@ -523,7 +557,170 @@ open class SBAUserDefaultsProfileItem: SBAProfileItemBase {
+        var storedVal = studyParticipant.value(forKey: key)
+        switch enumKey {
+        case .sharingScope:
+            guard let scopeString = storedVal as? String else { break }
+            storedVal = SBBParticipantDataSharingScope(key: scopeString)
+        default:
+            break
+        }
+        
+        return storedVal
+    }
+    
+    override open func setStoredValue(_ newValue: Any?) {
+        guard let studyParticipant = SBAStudyParticipantProfileItem.studyParticipant
+            else {
+                assertionFailure("Attempting to set \(sourceKey) (\(profileKey)) on nil SBBStudyParticipant")
+                return
+        }
+        guard !(sourceKey.hasPrefix("attributes.")) else {
+            let pathComponents = sourceKey.components(separatedBy: ".")
+            let attributeComponents = pathComponents[1...pathComponents.count - 1]
+            guard attributeComponents.count > 0 else {
+                assertionFailure("Error reading \(sourceKey) (\(profileKey)): \(sourceKey) is not a valid SBBStudyParticipant keypath")
+                return
+            }
+            let attributeKey = attributeComponents.joined(separator:".")
+            
+            super.setStoredValue(newValue, forKey: attributeKey)
+            return
+        }
+        guard let key = SBAProfileParticipantSourceKey(rawValue: sourceKey)
+            else {
+                assertionFailure("Error setting \(sourceKey) (\(profileKey)): \(sourceKey) is not a valid SBBStudyParticipant key")
+                return
+        }
+        
+        var setValue = newValue
+        switch key {
+        case .dataGroups:
+            guard let _ = newValue as? Set<String>
+                else {
+                    assertionFailure("Error setting \(sourceKey) (\(profileKey)): value \(String(describing: newValue)) cannot be converted to Set")
+                    return
+            }
+        case .sharingScope:
+            guard let scope = newValue as? SBBParticipantDataSharingScope
+                else {
+                    assertionFailure("Error setting \(sourceKey) (\(profileKey)): value \(String(describing: newValue)) cannot be converted to SBBParticipantDataSharingScope")
+                    return
+            }
+            setValue = SBBParticipantManager.dataSharingScopeStrings()[scope.rawValue]
+        case .notifyByEmail:
+            guard let _ = newValue as? Bool
+                else {
+                    assertionFailure("Error setting \(sourceKey) (\(profileKey)): value \(String(describing: newValue)) cannot be converted to Bool")
+                    return
+            }
+        default:
+            // the rest are String, and anything can be converted to String
+            break
+        }
+        
+        studyParticipant.setValue(setValue, forKeyPath: sourceKey)
+        
+        // save the change to Bridge
+        SBABridgeManager.updateParticipantRecord(studyParticipant) { (_, _) in }
```

Comment:
It will if the app is authenticated (has logged in and has a session token, not necessarily still valid). It won't during onboarding. I'm hoping edits to profile items stored here will be infrequent after that, and usually not done a bunch in a row. If that turns out not to be the case we can look at some kind of delay to gather a series of edits before firing off an update to the server, but this is simpler, easier to maintain, and more foolproof as far as making sure the updates all get to the server.